# Author: Damyan Ivanov <dmn@debian.org>
# Description: use proper Fedora location for lock files,
#              control sockets and such
--- src/jrd/file_params.h
+++ src/jrd/file_params.h
@@ -38,11 +38,11 @@ const int EVENT_VERSION		= 2;
 const int EVENT_SEMAPHORES	= 1;
 
 #ifdef NOHOSTNAME
-static const char* EVENT_FILE	= "isc_event1";
-static const char* LOCK_FILE	= "isc_lock1.gbl";
-static const char* INIT_FILE	= "isc_init1";
-static const char* GUARD_FILE	= "isc_guard1";
-static const char* MONITOR_FILE	= "isc_monitor1";
+static const char* EVENT_FILE	= "run/isc_event1";
+static const char* LOCK_FILE	= "run/isc_lock1.gbl";
+static const char* INIT_FILE	= "run/isc_init1";
+static const char* GUARD_FILE	= "run/isc_guard1";
+static const char* MONITOR_FILE	= "run/isc_monitor1";
 #elif defined(VMS)
 static const char* EVENT_FILE	= "[000000]isc_event1.%s";
 static const char* LOCK_FILE	= "[000000]isc_lock1.%s";
@@ -56,11 +56,11 @@ static const char* INIT_FILE	= "%s.int";
 static const char* GUARD_FILE	= "%s.grd";
 static const char* MONITOR_FILE	= "%s.mon";
 #else
-static const char* EVENT_FILE	= "isc_event1.%s";
-static const char* LOCK_FILE	= "isc_lock1.%s";
-static const char* INIT_FILE	= "isc_init1.%s";
-static const char* GUARD_FILE	= "isc_guard1.%s";
-static const char* MONITOR_FILE	= "isc_monitor1.%s";
+static const char* EVENT_FILE	= "run/isc_event1.%s";
+static const char* LOCK_FILE	= "run/isc_lock1.%s";
+static const char* INIT_FILE	= "run/isc_init1.%s";
+static const char* GUARD_FILE	= "run/isc_guard1.%s";
+static const char* MONITOR_FILE	= "run/isc_monitor1.%s";
 #endif
 
 #ifdef sun
--- configure.in
+++ configure.in
@@ -940,6 +940,7 @@ mkdir -p gen/firebird/examples/udf
 mkdir -p gen/firebird/lib
 mkdir -p gen/firebird/misc
 mkdir -p gen/firebird/help
+mkdir -p gen/firebird/run
 
 dnl #### TEMP ######### directories for .o and .d
 dnl # by module name
